spec/requests/students_spec.rb

```diff
@@ -0,0 +1,151 @@
+      end
+
+      it 'redirects to the created student' do
+        post students_url, params: { student: params }
+
+        expect(response).to redirect_to(student_url(Student.last))
+      end
+    end
+
+    context 'when invalid params are passed' do
+      let(:params) { { name: 'Ian', born_on: '' } }
+
+      it 'does not create a new a contact' do
+        expect { post('/students', params: { student: params }) }.not_to change(Student, :count)
+      end
+
+      it 'returns correct http status' do
+        post students_url, params: { student: params }
+
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
+
+      it 'renders correct template' do
+        post students_url, params: { student: params }
+
+        expect(response).to render_template(:new)
+      end
+    end
+  end
+
+  describe 'GET /show' do
+    let!(:student) { create(:student) }
+
+    it 'returns correct http status' do
+      get student_url(student)
+
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'renders correct template' do
+      get student_url(student)
+
+      expect(response).to render_template(:show)
+    end
+  end
+
+  describe 'GET /edit' do
+    let!(:student) { create(:student) }
+
+    it 'returns correct http status' do
+      get edit_student_url(student)
+
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'renders correct template' do
+      get edit_student_url(student)
+
+      expect(response).to render_template(:edit)
+    end
+  end
+
+  describe 'PATCH /update' do
+    let!(:student) { create(:student) }
+
+    context 'when valid params are passed' do
+      let(:params) { { name: 'Ian', born_on: Date.new(1998, 10, 16) } }
+
+      it 'updates a student' do
+        expect { patch student_url(student), params: { student: params } }.to change { student.reload.name }.to('Ian')
+      end
+
+      it 'redirects to the student' do
+        patch student_url(student), params: { student: params }
+
+        expect(response).to redirect_to(student_url(student))
+      end
+    end
+
+    context 'when invalid params are passed' do
+      let(:params) { { name: 'Ian', born_on: '' } }
+
+      it 'does not update the student' do
+        expect { patch student_url(student), params: { student: params } }.to_not(change { student.reload.name })
+      end
+
+      it 'renders correct http status' do
+        patch student_url(student), params: { student: params }
+
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
+
+      it 'renders correct template' do
+        patch student_url(student), params: { student: params }
+
+        expect(response).to render_template(:edit)
+      end
+    end
+  end
+
+  describe 'DELETE /destroy' do
+    let!(:student) { create(:student) }
+
+    it 'destroys the requested contact' do
+      expect { delete student_url(student) }.to change(Student, :count).by(-1)
+    end
```

Comment:
`[Sugestão]` 

Sei que esse teste foi parecido com o que nos foi apresentado, mas particularmente acho ele um pouco vago, principalmente quando utilizado sozinho. A contagem de `Student` mudar por -1 não quer dizer, necessariamente, que o `requested student was destroyed`, mas sim que a contagem de `Students` foi reduzida em uma unidade. Sendo assim, qualquer `student` poderia ter sumido. Eu repensaria esse teste mais ou menos assim (qualquer uma das expectations serviria, não consegui decidir qual seria melhor):

```suggestion
    it 'destroys the requested student' do
      delete student_url(student)
      
      expect(Student.find_by_id(student)).to be_nil
      expect { Student.find(student.id) }.to raise_error(ActiveRecord::RecordNotFound)
      expect { student.reload }.to raise_error(ActiveRecord::RecordNotFound)
      expect(Student.where(id: student.id)).to be_empty
    end
```